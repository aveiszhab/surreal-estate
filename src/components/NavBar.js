import React from "react";
import "../styles/NavBar.css";
import { Link } from "react-router-dom";
import MyLogo from "../MyLogo.png";

const NavBar = () => {
  return (
    <div className="navbar">
      <img className="navbar-logo" src={MyLogo} alt="logo" />
      <h2>Surreal Estate</h2>
      <ul className="navbar-links">
        <li className="navbar-links-item">
          <Link to="/">View Properties</Link>
        </li>
        <li className="navbar-links-item">
          <Link to="/add-property">Add Property Page</Link>
        </li>
      </ul>
    </div>
  );
};

export default NavBar;
