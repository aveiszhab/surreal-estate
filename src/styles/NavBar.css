.navbar{
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    background-color: white;
    align-items: center;
}

h2 {
    text-align: center;
    font-weight: 900;
}

.navbar-logo{
    width: 10%;    
    margin-right: 2%;
}

.navbar-links{
    list-style: none;
    display: flex;
    flex-wrap: nowrap;
    flex:1;
    margin: 0px;
    justify-content: center ;
    padding-inline-start: 0px;
}

.navbar-links-item {
    padding: 0% 4%;
    font-weight: 800;
    text-align: center;
}
.navbar-links-item :hover {
    color: rgb(16, 58, 16)
}

.navbar-links-item > * {
    text-decoration: none;
    color: black;
}

@media only screen and (max-width: 600px) {
    [class*="navbar-logo"]{
        width: 20%
    }
}
