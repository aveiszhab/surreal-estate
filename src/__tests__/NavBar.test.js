import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  const setUp = () => {
    const { asFragment, getByRole } = render(
      <Router>
        <NavBar />
      </Router>
    );

    return {
      asFragment: asFragment(),
      logo: getByRole("img", { name: /logo/i }),
      linkToProperties: getByRole("link", {
        name: /View Properties/i,
      }),
      linkToAddProperty: getByRole("link", {
        name: /Add Property/i,
      }),
    };
  };

  let renderedComponent;

  beforeEach(() => {
    renderedComponent = setUp();
  });

  it("renders correctly to match the NavBar snapshot", () => {
    expect(renderedComponent.asFragment).toMatchSnapshot();
  });

  it("renders logo correctly", () => {
    expect(renderedComponent.logo).toHaveClass("navbar-logo");
    expect(renderedComponent.logo).toHaveAttribute(
      "src",
      expect.stringContaining("MyLogo")
    );
    expect(renderedComponent.logo).toHaveAttribute(
      "alt",
      expect.stringContaining("logo")
    );
  });

  it("renders properties link properly", () => {
    expect(renderedComponent.linkToProperties).toHaveAttribute("href", "/");
  });

  it("renders add property link properly", () => {
    expect(renderedComponent.linkToAddProperty).toHaveAttribute(
      "href",
      "/add-property"
    );
  });
});
