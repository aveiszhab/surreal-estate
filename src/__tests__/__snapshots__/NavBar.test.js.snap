// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly to match the NavBar snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="logo"
      class="navbar-logo"
      src="MyLogo.png"
    />
    <h2>
      Surreal Estate
    </h2>
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        <a
          href="/"
        >
          View Properties
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          href="/add-property"
        >
          Add Property Page
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
